substrate/client/network/sync/src/engine.rs

```diff
@@ -787,29 +792,44 @@ where
 				return
 			}
 
+			// reset all timers for peers so that they're not evicted immediately
+			// after the initial wait period is over
+			for context in self.peers.values_mut() {
+				context.last_notification_received = Instant::now();
+				context.last_notification_sent = Instant::now();
+			}
+
 			self.syncing_started = None;
-			self.last_notification_io = Instant::now();
 		}
 
-		// if syncing hasn't sent or received any blocks within `INACTIVITY_EVICT_THRESHOLD`,
-		// it means the local node has stalled and is connected to peers who either don't
-		// consider it connected or are also all stalled. In order to unstall the node,
-		// disconnect all peers and allow `ProtocolController` to establish new connections.
-		if self.last_notification_io.elapsed() > INACTIVITY_EVICT_THRESHOLD {
-			log::debug!(
-				target: LOG_TARGET,
-				"syncing has halted due to inactivity, evicting all peers",
-			);
+		// go over all connected peers and check if any of them have been idle for a while. Idle
```

Comment:
This comment is talking about peers that have silently disconnected to us and SyncingEngine does not know about it. However, in cases where the chain is not advancing because block production stopped, we would still rotate all the peers at the same time right? And maybe even the whole network would do it at the same time because the block announcements stopped. Also with coretime, a parachain might not have a core all the time, so it would be normal to not receive block announcements for a while, might be a case to consider.

Reply:
Very good points. The burn-in on Kusama AssetHub a failure as well so I don't think this PR will be merged.